File: services/backend/src/core_service/modules/schedule/domain/schedule_item.py
"""ScheduleItem 도메인 엔티티 — schema.md §5 `schedule_items` 테이블 매핑.

⚠️ 골격 단계: Repository/Service/API 미구현.
"""

from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum
from uuid import UUID


class ScheduleKind(StrEnum):
    APPOINTMENT = "appointment"
    MEDICATION = "medication"


class ScheduleStatus(StrEnum):
    PENDING = "pending"
    CONFIRMED = "confirmed"
    MISSED = "missed"
    DECLINED = "declined"


@dataclass
class ScheduleItem:
    id: UUID
    user_id: UUID
    kind: ScheduleKind
    description: str | None
    location: str | None
    recurrence: str | None
    due_at: datetime
    status: ScheduleStatus
    remind_count: int
    next_remind_at: datetime | None
    decline_reason: str | None
    responded_at: datetime | None

File: apps/web/src/types/schedule-item.ts
/** design.md §3.1 ScheduleItem — schema.md `schedule_items` 매핑. */
export type ScheduleKind = "appointment" | "medication";
export type ScheduleStatus = "pending" | "confirmed" | "missed" | "declined";

export interface ScheduleItem {
  id: string;
  userId: string;
  kind: ScheduleKind;
  description?: string;
  location?: string;
  recurrence?: string;
  dueAt: string;
  status: ScheduleStatus;
  remindCount: number;
  declineReason?: string;
  nextRemindAt?: string;
  respondedAt?: string;
}
